Statistics/Test/Levene.hs

```diff
@@ -0,0 +1,144 @@
+Assesses equality of variances, robust to non-normality, and versatile with mean or median centering.
+-}
+module Statistics.Test.Levene (
+    Center(..),
+    TestResult(..),
+    levenesTest
+) where
+
+import qualified Data.Vector as V
+import qualified Data.Vector.Unboxed as U
+import qualified Data.Vector.Algorithms.Merge as VA
+import Statistics.Distribution
+import Statistics.Distribution.FDistribution (fDistribution)
+import qualified Statistics.Sample as Sample
+import Control.Exception (assert)
+
+-- | Center calculation method with proper documentation
+data Center = 
+    Mean      -- ^ Use arithmetic mean
+  | Median    -- ^ Use median
+  | Trimmed Double  -- ^ Trimmed mean with given proportion to cut from each end
+  deriving (Eq, Show)
+
+-- | Result type with comprehensive information
+data TestResult = TestResult {
+    wStatistic :: Double    -- ^ Levene's W statistic
+  , pValue     :: Double    -- ^ Associated p-value
+  , rejectNull :: Bool      -- ^ Whether to reject null hypothesis at given alpha
+  } deriving (Eq, Show)
+
+-- | Trim data from both ends with error handling and performance optimization
+trimboth :: (Ord a, Fractional a) 
+         => V.Vector a 
+         -> Double 
+         -> Either String (V.Vector a)
+trimboth vec prop
+  | prop < 0 || prop > 1 = Left "Proportion must be between 0 and 1"
+  | V.null vec = Right vec
+  | otherwise = do
+      let sorted = V.modify VA.sort vec
+          n = V.length sorted
+          lowerCut = floor (prop * fromIntegral n)
+          upperCut = n - lowerCut
+      assert (upperCut >= lowerCut) $ 
+        Right $ V.slice lowerCut (upperCut - lowerCut) sorted
+
+-- | Calculate median using pre-sorted vector
+vectorMedian :: (Fractional a, Ord a) 
+             => V.Vector a 
+             -> Either String a
+vectorMedian vec
+  | V.null vec = Left "Empty vector in median calculation"
+  | otherwise = Right $ 
+      if odd len 
+        then sorted V.! mid 
+        else (sorted V.! (mid - 1) + sorted V.! mid) / 2
+  where
+    sorted = V.modify VA.sort vec
+    len = V.length sorted
+    mid = len `div` 2
+
+-- | Main Levene's test function with full error handling
+levenesTest :: Double       -- ^ Significance level (alpha)
+            -> Center       -- ^ Center calculation method
+            -> [V.Vector Double] -- ^ Input samples
+            -> Either String TestResult
+levenesTest alpha center samples
+  | alpha < 0 || alpha > 1 = Left "Significance level must be between 0 and 1"
+  | length samples < 2 = Left "At least two samples required"
+  | otherwise = do
+      processed <- mapM processSample samples
+      let (deviations, ni) = unzip processed
+          k = length samples
+          n = sum ni
+          zbari = map Sample.mean deviations
+          zbar = sum (zipWith (*) zbari (map fromIntegral ni)) / fromIntegral n
+
+          numerator = sum [fromIntegral (ni !! i) * (zbari !! i - zbar) ** 2 | i <- [0..k-1]]
```

Comment:
`!!` is generally smell since it's O(n). `zbari` and `ni` should probably be vector